### PR TITLE
Issues #147 & #148: Removed Who's Online & Who's New blocks.

### LINKED ISSUE #147: Remove "Who's new" block from core
Relevant drupal.org issue: https://drupal.org/node/2020395

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/147 & https://github.com/backdrop/backdrop-issues/issues/148

### CODE REVIEW

**Comment by ghost** [comment]
Not sure what's wrong with this... Any ideas @quicksketch?

**Comment by quicksketch** [comment]
Looks like there's a parse error in the code itself. Travis CI reports:

```
PHP Parse error:  syntax error, unexpected T_SL in /home/travis/build/backdrop/backdrop/core/modules/user/user.module on line 967
```

I think that's the `<` character, it probably indicates an unresolved conflict that was committed.

**Comment by ghost** [comment]
Oops, left some merge tags in there somehow. Removed.

File: core/modules/layout/tests/layout.test [hunk 1]
```diff
@@ -1102,16 +1096,6 @@ class LayoutUpgradePathTest extends UpgradePathTestCase {
...
-
-    // Logout, and check the homepage for the Who's new blocks, which should
-    // only be visible to authenticated users.
-    $this->backdropLogout();
-    $this->backdropGet('<front>');
-    $this->assertNoBlocks(array(
-      '.region-triptych-first' => array(
-        'block-user-new',
-      ),
-    ));
```
Comment (quicksketch) [inline core/modules/layout/tests/layout.test:1114]: We need to replace these test with new ones to ensure user roles are upgraded properly. We can use any block we like, but it still needs to be tested.

The block list that gets upgraded is specified in the layout_upgrade_database.php file, that should be updated to use different blocks and then this test should be made to match.

**Comment by ghost** [comment]
Ok, try this on for size. Will rebase before committing.

**Pull request merged**